src/util/psbt/mod.rs

```diff
@@ -209,6 +218,473 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Attempts to create _all_ the required signatures for this PSBT using `k`.
+    ///
+    /// If you just want to sign an input with one specific key consider using `sighash_ecdsa`. This
+    /// function does not support scripts that contain `OP_CODESEPARATOR`.
+    ///
+    /// # Panics
+    /// If an attempt is made to sign a taproot input.
+    pub fn sign<C: Signing>(&mut self, k: impl GetKey + Copy, secp: &Secp256k1<C>) -> Result<(), SignError> {
+        // FIXME: Is cloning the best we can do here?
```

Comment:
Hmm, yeah I can't seem to come up with something better. We do a mutable borrow at `self.bip32_sign_ecdsa()` below.

Reply:
Cheers, removed the comment.